plugins/outputs/redis/redis.go

```diff
@@ -0,0 +1,117 @@
+package redis
+
+import (
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"gopkg.in/redis.v4"
+)
+
+//use the redis service LIST struct as telegraf output
+type RedisOutput struct {
+	Server      string            `toml:"server"`
+	Password    string            `toml:"password"`
+	IdleTimeout internal.Duration `toml:"idle_timeout"`
+	Timeout     internal.Duration `toml:"timeout"`
+	Queue       string            `toml:"queue_name"`
+
+	server     *redis.Client
+	serializer serializers.Serializer
+}
+
+var sampleConfig = `
+  ## redis service listen addr:port, default 127.0.0.1
+  # server = "127.0.0.1:6379"
+  ## redis service login password
+  # password = ""
+  ## redis close connections after remaining idle for this duration.
+  ## if the value is zero, then idleconnections are not closed.
+  ## shoud set the timeout to a value lessthan the redis server's timeout.
+  # idle_timeout = "1s" 
+  ## specifies the timeout for reading/writing a single command.
+  # timeout = "1s"
+  ## redis list name, defalut telegraf/output
+  # queue_name = "telegraf/output"
```

Comment:
Since redis has several ways to do queuing, and will be adding more including [streams](http://antirez.com/news/114), I think we should prepare for supporting other output methods in the future.

To avoid any confusion with other ways of outputting, I think we should refer to this specific structure as a list in the configuration and in the code.

We have also used a style of configuration in a few other plugins that look like this:
```toml
[[outputs.redis]]
  # ... server / connection config ...
  [outputs.redis.strategy]
    method = "list"
    key = "telegraf/output"
```

The advantages of this style over having parameters top level are:
- Can only have one strategy and be valid toml, which is a hint about what can be done.
- Scales up nicely if for some reason you need another option

The downside is that is a little more complicated and people get tripped up on toml syntax sometimes.